scipy/linalg/flapack_sym_herm.pyf.src

```diff
@@ -302,6 +302,46 @@ end subroutine <prefix2c>heevd_lwork
    end subroutine <prefix>sytrf_lwork
 
 
+   subroutine <prefix>sytrs(n,nrhs,a,lda,ipiv,b,ldb,info,lower)
+   
+   ! Solve A * X = B for symmetric A matrix after calling ?sytrf
+   
+     callstatement (*f2py_func)((lower?"L":"U"),&n,&nrhs,a,&lda,ipiv,b,&ldb,&info)
+     callprotoargument char*,F_INT*,F_INT*,<ctype>*,F_INT*,F_INT*,<ctype>*,F_INT*,F_INT*
+     
+     <ftype> intent(in), dimension(lda, n), check((lda >= n) && (n >= 0)) :: a
+     <ftype> intent(in,out,copy,out=x),dimension(ldb, nrhs) :: b
+     integer optional,intent(in),check(lower==0||lower==1) :: lower = 0
+     integer intent(in),dimension(n),depend(n) :: ipiv
+     integer intent(hide), depend(a),check(n >= 0) :: n = shape(a, 1)
+     integer intent(hide), depend(b) :: nrhs = shape(b, 1)
+     integer intent(hide), depend(a) :: lda = MAX(1, shape(a, 0))
+     integer intent(hide), depend(b,n),check(ldb >= n) :: ldb = MAX(1, shape(b, 0))
+     integer intent(out):: info
+
+   end subroutine <prefix>sytrs
+
+
+   subroutine <prefix2c>hetrs(n,nrhs,a,lda,ipiv,b,ldb,info,lower)
+   
+   ! Solve A * X = B for hermitian A matrix after calling ?hetrf
+   
+     callstatement (*f2py_func)((lower?"L":"U"),&n,&nrhs,a,&lda,ipiv,b,&ldb,&info)
+     callprotoargument char*,F_INT*,F_INT*,<ctype2c>*,F_INT*,F_INT*,<ctype2c>*,F_INT*,F_INT*
+     
+     <ftype2c> intent(in), dimension(lda, n), check((lda >= n) && (n >= 0)) :: a
+     <ftype2c> intent(in,out,copy,out=x),dimension(ldb, nrhs) :: b
```

Comment:
I think I get the idea of `(in,out,copy,out=x)` - there's no separate `x` input just to store the output, so this argument is both an input and an output, and to avoid modifying the input, we copy it. But how should I parse the syntax?

Reply:
`intent(in, out)` or in the old syntax `intent(inout)` was to overwrite the variable. if you also add `copy` then it automatically adds a kwarg `overwrite_b` in case you want to do that. `out` defines the output name. If you omit it, you'll see the output variable name `b`. This is just to make a distinction in the docstring.